ext/vulnsrc/redhat/ovalv2.go

```diff
@@ -0,0 +1,441 @@
+	}
+	return ovalV2DefinitionNamespaces
+}
+
+// parse one definition
+func ParseCriteriaForModuleNamespaces(ovalDefinition OvalV2Definition) ([]string, error) {
+    var moduleNamespaces []string
+	criteria := extractAllCriterions(ovalDefinition.Criteria)
+	// walk the criteria and add them
+	for _, criterion := range criteria {
+		// Module idm:DL1 is enabled
+		var regexComment = regexp.MustCompile(`(Module )(.*)( is enabled)`)
+		matches := regexComment.FindStringSubmatch(criterion.Comment)
+		if matches != nil && len(matches) > 2 && matches[2] != "" {
+			moduleNamespaces = append(moduleNamespaces, matches[2])
+		}
+		// moduleNamespaces = append(moduleNamespaces, criterion.Comment)
+	}
+    return moduleNamespaces, nil
+}
+
+func extractAllCriterions(criteria OvalV2Criteria) []OvalV2Criterion {
+    var criterions []OvalV2Criterion
+    for _, criterion := range criteria.Criteria {
+        // recursively append criteria contents
+        criterions = append(criterions, extractAllCriterions(criterion)...)
+    }
+    for _, criterion := range criteria.Criterion {
+        // append criterion
+        criterions = append(criterions, criterion)
+    }
+    return criterions
+}
+
+func ParseNVRA(rpmName string) RpmNvra {
+	// parse NVRA from RPM name
+	// golang-1.6.3-2.el7.x86_64.rpm
+	// Name        : golang
+	// Version     : 1.6.3
+	// Release     : 2.el7
+	// Architecture: x86_64
+	var regexRpmNVRA = regexp.MustCompile(`(.*/)*(.*)-(.*)-(.*?)\.([^.]*)(\.rpm)`)
+	matches := regexRpmNVRA.FindStringSubmatch(rpmName)[2:6]
+	rpmNvra := RpmNvra{matches[0], matches[1], matches[2], matches[3]}
+	return rpmNvra
+}
+
+func IsRmpArchSupported(rpmName string) bool {
+	return IsArchSupported(ParseNVRA(rpmName).Arch)
+}
+
+func IsArchSupported(arch string) bool {
+	return SupportedArches[arch]
+}
+
+// parse affected_cpe_list (first entry from CPE list should not be used because it doesn't come from Advisory configuration)
+func ParseParseCpeNameFromAffectedCpeList(affectedCpeList OvalV2Cpe) ([]CpeName, error) {
+	cpeNames := []CpeName{}
+	if affectedCpeList.Cpe == nil || len(affectedCpeList.Cpe) < 2 {
+		return cpeNames, errors.New("unparseable affected cpe list")
+	}
+	// parse and return any entries after the first cpe entry from the list
+	for i := 1; i < len(affectedCpeList.Cpe); i++ {
+		cpeNames = append(cpeNames, ParseCpeName(affectedCpeList.Cpe[i]))
+	}
+	return cpeNames, nil
+}
+
+// parse cpe string
+func ParseCpeName(cpeNameString string) CpeName {
+	// cpe:/ {part} : {vendor} : {product} : {version} : {update} : {edition} : {language}
+	// remove the "cpe:/" prelude and split the cpe name string into its components
+	components := strings.Split(strings.Replace(cpeNameString, "cpe:/", "", 1), ":")
+	// components slice must contain 7 elements; append empty string to any missing elements
+	for i := len(components); i < 7; i++ {
+		components = append(components, "")
+	}
+	return CpeName{
```

Comment:
Does CPE always have all these componsens? Is there any chance that cpe will have more or less?